lib/compiler/passes/generate-js.js

```diff
@@ -788,15 +788,13 @@ function generateJS(ast, options) {
       "}",
       "",
       "function peg$SyntaxError(message, expected, found, location) {",
-      "  this.message = message;",
-      "  this.expected = expected;",
-      "  this.found = found;",
-      "  this.location = location;",
-      "  this.name = \"SyntaxError\";",
-      "",
-      "  if (typeof Error.captureStackTrace === \"function\") {",
-      "    Error.captureStackTrace(this, peg$SyntaxError);",
-      "  }",
+      "  const self = Error.call(this, message);",
+      "  Object.setPrototypeOf(self, peg$SyntaxError.prototype);",
```

Comment:
See https://webreflection.co.uk/blog/2016/08/30/js-super-problem
What I want is for `peg$syntaxError` to be a real subclass of Error, with the constructor having been called correctly.  Below this you see a call to `peg$subclass` which only does part of that.